core/arch/aarch64/instr_create.h

```diff
@@ -271,13 +357,27 @@
     opnd_create_reg_ex(opnd_get_reg(rm_or_imm), 0, DR_OPND_SHIFTED), \
     opnd_add_flags((sht), DR_OPND_IS_SHIFT), (sha)) : \
   instr_create_1dst_4src((dc), OP_add, (rd), (rn), (rm_or_imm), (sht), (sha))
-#define XINST_CREATE_and(dc, d, s) \
-  INSTR_CREATE_and_shift((dc), (d), (d), (s), \
-    OPND_CREATE_INT8(DR_SHIFT_LSL), OPND_CREATE_INT8(0))
+#define INSTR_CREATE_adds(dc, rd, rn, rm_or_imm) \
```

Comment:
Yes, this is what I was referring to up above: so you have it for some instructions, it just needs to be added for INSTR_CREATE_and.